pkg/controllers/addoninstall/addon_install_controller_test.go

```diff
@@ -0,0 +1,168 @@
+package addoninstall
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	kubescheme "k8s.io/client-go/kubernetes/scheme"
+	"open-cluster-management.io/addon-framework/pkg/addonmanager/constants"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	scheme                 = runtime.NewScheme()
+	logCertSecretNamespace = "open-cluster-management"
+	logCertSecretName      = "ocm-klusterlet-self-signed-secrets"
+)
+
+func newTestReconciler(addOnName string, existingObjs ...runtime.Object) *Reconciler {
+	s := kubescheme.Scheme
+	clusterv1.AddToScheme(s)
```

Comment:
AddToScheme is deprecated, please use `Install`
```suggestion
	clusterv1.Install(s)
	addonapiv1alpha1.Install(s)
```

Reply:
Updated.